src/components/SelectOrAddNewAddressCard.vue

```diff
@@ -258,23 +296,77 @@ const addresses = computed(() => {
         return
     }
 
-    return nodes.map(
-        (address) =>
-            new AddressImpl(
-                address.id,
-                address.street1,
-                address.street2,
-                address.city,
-                address.postalCode,
-                address.country,
-                address.companyName ?? undefined
-            )
-    )
+    return nodes.map((address) => {
+        let name: undefined | Name = undefined
+        if (address.name) {
+            name = new NameImpl(address.name.firstName, address.name.lastName)
+        }
+
+        return new AddressImpl(
+            address.city,
+            address.country,
+            address.id,
+            address.postalCode,
+            address.street1,
+            address.street2,
+            address.companyName ?? undefined,
+            name
+        )
+    })
 })
 
 const showExpansion = ref<boolean>(false)
 
-const name = ref<string | null | undefined>(null)
+const firstName = ref<string | null | undefined>(null)
+const lastName = ref<string | null | undefined>(null)
+
+/**
+ * Whether both refs, `firstName` and `lastName`, are undefined.
+ */
+const nameIsUndefined = computed(() => {
+    return firstName.value == undefined && lastName.value == undefined
+})
+
+/**
+ * Validates the first name based on the presence of a last name.
+ *
+ * This function checks if the `lastName` field has a non-empty value. If it does,
+ * it ensures that the `firstName` field also has a non-empty value. If the `firstName`
+ * field is empty while the `lastName` field is not, it returns an error message.
+ *
+ * @returns Returns `true` if the validation passes, or a string
+ * indicating the validation error if it fails.
+ */
+function firstNameIsValid(): boolean | string {
+    if (isNonEmptyString(lastName.value)) {
+        return isNonEmptyString(firstName.value)
```

Comment:
you could move this condition to the if (with an &&), would imho improve readability

Reply:
Will do.

Reply:
Wait, I don't see how this would still be the same code. Because I only want to return the error message if the first condition was true and the second was not.

Reply:
oh right
you of course would need to negate the second thing

Reply:
fc99c6413841bcc99cae81966a52a76faba11d9e